dds/DCPS/transport/framework/TransportReceiveStrategy_T.cpp

```diff
@@ -78,6 +78,173 @@ TransportReceiveStrategy<TH, DSH>::check_header(const DSH& /*header*/)
   return true;
 }
 
+template<typename TH, typename DSH>
+int
+TransportReceiveStrategy<TH, DSH>::handle_simple_dds_input(ACE_HANDLE fd)
+{
+  DBG_ENTRY_LVL("TransportReceiveStrategy", "handle_simple_dds_input", 6);
+
+  for (int index = 0; index < RECEIVE_BUFFERS; ++index) {
+    if (receive_buffers_[index] == 0) {
+      ACE_NEW_MALLOC_RETURN(
+        receive_buffers_[index],
+        (ACE_Message_Block*) mb_allocator_.malloc(sizeof(ACE_Message_Block)),
+        ACE_Message_Block(
+          RECEIVE_DATA_BUFFER_SIZE,           // Buffer size
+          ACE_Message_Block::MB_DATA,         // Default
+          0,                                  // Start with no continuation
+          0,                                  // Let the constructor allocate
+          &data_allocator_,                   // Our buffer cache
+          &receive_lock_,                     // Our locking strategy
+          ACE_DEFAULT_MESSAGE_BLOCK_PRIORITY, // Default
+          ACE_Time_Value::zero,               // Default
+          ACE_Time_Value::max_time,           // Default
+          &db_allocator_,                     // Our data block cache
+          &mb_allocator_                      // Our message block cache
+        ),
+        -1);
+    }
+  }
+
+  ACE_Message_Block* cur_rb = receive_buffers_[buffer_index_];
+  iovec iov;
+#ifdef _MSC_VER
+#pragma warning(push)
+// iov_len is 32-bit on 64-bit VC++, but we don't want a cast here
+// since on other platforms iov_len is 64-bit
+#pragma warning(disable : 4267)
+#endif
+  iov.iov_len = cur_rb->space();
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+  iov.iov_base = cur_rb->wr_ptr();
+
+  ACE_INET_Addr remote_address;
+  bool stop = false;
+  ssize_t bytes_remaining = receive_bytes(&iov,
+                                          1,
+                                          remote_address,
+                                          fd,
+                                          stop);
+
+  cur_rb->wr_ptr(bytes_remaining);
+
+  if (stop) {
+    cur_rb->reset();
```

Comment:
Instead of needing a `reset()` in a few different places, could it just be a single `reset()` right after the initialization of `cur_rb`?